apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,206 @@
+        // model, not their JavaScript-friendly equivalents.  The mapping
+        // occurs in the constructor, below, and back when we submit.
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        state: PropTypes.string,
+        zip: PropTypes.string,
+        full_address: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    onClose: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    onClose: function () {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    const {existingSchoolInfo} = this.props.scriptData;
+    const initialNcesSchoolId = existingSchoolInfo.school_id ?
+      existingSchoolInfo.school_id :
+      (
+        existingSchoolInfo.country === 'United States'
+        &&
+        SCHOOL_TYPES_HAVING_NCES_SEARCH.includes(existingSchoolInfo.school_type)
+        &&
+        (existingSchoolInfo.school_name || existingSchoolInfo.state || existingSchoolInfo.zip)
+      ) ? '-1' : '';
+
+    this.state = {
+      country: existingSchoolInfo.country || '',
+      schoolType: existingSchoolInfo.school_type || '',
+      schoolName: existingSchoolInfo.school_name || '',
+      schoolState: existingSchoolInfo.state || '',
+      schoolZip: existingSchoolInfo.zip || '',
+      schoolLocation: existingSchoolInfo.full_address || '',
+      ncesSchoolId: initialNcesSchoolId,
+      showSchoolInfoUnknownError: false,
+    };
+  }
+
+
+  handleSchoolInfoSubmit = () => {
+    let schoolData;
+    if (this.state.ncesSchoolId === '-1') {
+      schoolData = {
+        "user[school_info_attributes][school_name]": this.state.schoolName,
+        "user[school_info_attributes][school_state]": this.state.schoolState,
+        "user[school_info_attributes][school_zip]": this.state.schoolZip,
+        "user[school_info_attributes][full_address]": this.state.schoolLocation,
+      };
+    } else {
+      schoolData = {
+        "user[school_info_attributes][school_id]": this.state.ncesSchoolId,
+      };
+    }
+
+    const {formUrl, authTokenName, authTokenValue} = this.props.scriptData;
+    $.post({
+      url: formUrl + '.json',
+      dataType: "json",
+      data: {
+        '_method': 'patch',
+        [authTokenName]: authTokenValue,
+        "user[school_info_attributes][country]": this.state.country,
+        "user[school_info_attributes][school_type]": this.state.schoolType,
+        ...schoolData,
+      },
+    }).done(() => {
+      this.props.onClose();
+    }).fail(() => {
+      if (!this.state.showSchoolInfoUnknownError) {
+        // First failure, display error message and give the teacher a chance
+        // to try again.
+        this.setState({showSchoolInfoUnknownError: true});
+      } else {
+        // We already failed once, let's not block the teacher any longer.
+        // TODO (bbuchanan): Gather metrics on this.
+        this.props.onClose();
+      }
+    });
```

Comment:
Is this unknown error because we don't expect any validation to run on the submission?

Reply:
This is a a fallback in case of an unexpected error response on submission - likely caused by the service being temporarily unavailable, so we ask the teacher to try again at least once.  Based on this handler in the Census Teacher Banner:

https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/census2017/CensusTeacherBanner.jsx#L194-L199